lib/commands/deploy.ts

```diff
@@ -3,12 +3,23 @@
 
 export class DeployOnDeviceCommand implements ICommand {
 	constructor(private $platformService: IPlatformService,
-		private $platformCommandParameter: ICommandParameter) { }
+		private $platformCommandParameter: ICommandParameter,
+				private $options: IOptions,
+				private $errors: IErrors) { }
 
 	execute(args: string[]): IFuture<void> {
 		return this.$platformService.deployOnDevice(args[0]);
 	}
 
+	public canExecute(args: string[]): IFuture<boolean> {
+		return (() => {
+			if (this.$options.release && (!this.$options.keyStorePath || !this.$options.keyStorePassword || !this.$options.keyStoreAlias || !this.$options.keyStoreAliasPassword)) {
+				this.$errors.fail("When producing a release build, you need to specify all --key-store-* options.");
+			}
+			return true;
+		}).future<boolean>()();
+	}
+
 	allowedParameters = [this.$platformCommandParameter];
```

Comment:
you should change allowedParameters to be:
```TypeScript
allowedParameters = [];
```
in order to avoid confusion, as it will never be used.